2016/proposals/chris-medrela.md

```diff
@@ -0,0 +1,560 @@
+# Manage workflow for Software/Data Carpentry instructor training
+
+## Abstract
+
+Software Carpentry (SWC) has a workflow for Instructor Trainings. Currently, it uses Etherpads and Google Docs spreadsheets, which slows down SWC Admins and Trainers whose time is already precious. The goal is to extend AMY (Django-base web application for SWC Admins) to manage the process of training people to be instructors, to avoid manual data copying and synchronization and make the entire process smoother.
+
+[The appendix](#appendix) at the end of this proposal contains [description of the current workflow] and [list of issues] based on [Greg Wilson's pull request] and its discussion. Also, whenever something is written Title-cased, you can look it up in the [glossary](#glossary) in the [appendix](#appendix) at the end of this proposal. Hopefully, the glossary will keep this proposal unambiguous. It also serves as a list of all [user roles] and [resources](#resources). The latter is also based on the Greg's pull request.
+
+[Greg Wilson's pull request]: https://github.com/swcarpentry/board/pull/81/files
+[user roles]: people-or-user-roles
+
+Here is [#89](https://github.com/numfocus/gsoc/issues/89) issue where I discussed this proposal. 
+
+I split this proposal into as small milestones as possible. Every milestone ends up with new version of Amy being pushed to production and introduced to all its users. This minimizes the chances of failure and let us adapt or change almost everything after every milestone -- even if it means changing the scope of this project. The milestones are:
```

Comment:
@gvwilson, what about splitting the first milestone into two stages? In the first stage, we could move only the Final Demonstration Session into Amy, so trainees register for Discussion Session on Etherpad, then create an account in Amy and then register for Final Demonstration Session in Amy. In the second stage, I'd implement the rest -- that is Homework and registration for Discussion Session. The first stage would take about 5 weeks, and the second one about 3 weeks, instead of one 8-weeks long milestone. I'm not sure about this idea since it could create an additional overhead for Admins and Trainers during second stage.